### ISSUE TITLE
[BUG] forecaster fit/predict interface

### ISSUE THREAD

**Issue by mloning** [alan-turing-institute/sktime#1649]
**Describe the bug**
I think the current behaviour of the forecasting API isn't very consistent. My preferred specification is described here: https://github.com/sktime/community-org/blob/main/meetups/previous_meetings/20210909-meetup.md

**To Reproduce**
```python
import numpy as np
from sktime.forecasting.exp_smoothing import ExponentialSmoothing
from sktime.forecasting.naive import NaiveForecaster
from sktime.forecasting.model_evaluation import evaluate
from sktime.forecasting.model_selection import SlidingWindowSplitter
from sktime.utils.plotting import plot_series

# fails with: TypeError: No valid mtype could be identified
y = np.random.normal(size=20)
forecaster = ExponentialSmoothing()
forecaster.fit(y)

# works but predict returns np.ndarray
y = np.random.normal(size=(20, 1))
forecaster = NaiveForecaster()
forecaster.fit(y)
y_pred = forecaster.predict(fh=1)
type(y_pred)
# numpy.ndarray

# but then other functionality breaks
# fails with TypeError
plot_series(y_pred)

# fails with TypeError: No valid mtype could be identified
y = np.random.normal(size=20)
forecaster = NaiveForecaster()
forecaster.fit(y)

# fails with ValueError: Data must be 1-dimensional
y = np.random.normal(size=(20, 1))
forecaster = ExponentialSmoothing()
forecaster.fit(y)

# fails with ValueError: Index data must be 1-dimensional
y = np.random.normal(size=(20, 1))
forecaster = NaiveForecaster()
cv = SlidingWindowSplitter(fh=1)
evaluate(forecaster, cv, y)
```

**Expected behavior**
Consistent handling of numpy arrays. User-friendly error messages. Preferably also consistent output types.

**Comment by fkiraly** [comment]
@mloning, some of this was one of the things fixed in the transformers refactor, #1365.
Originally, we were going to allow only 2D numpy, but decided to relax to 1D numpy
With it merged, the checks for 1D now work properly, here;s the relevant bullet point:

> relaxing np.ndarray mtype for Series so 1D numpy arrays are also allowed (and considered univariate series or vector). This is required since some tests use univariate numpy arrays as label vectors.

On top of this, there was another genuine bug (thanks for reporting!), namely:
`_split` and `convert` would break since both were calling `pandas` constructors for `Series` and `Index` with 2D `np.ndarray` which can break, depending on `pandas` version. Similar to problems I believe we have with older `pandas` version.
I fixed this by adding a `flatten` in the right places.

**Comment by mloning** [comment]
Update is also broken:

```python
y = pd.Series(np.random.normal(size=20))
forecaster = NaiveForecaster()
forecaster.fit(y)
y_pred = forecaster.predict(fh=1)
forecaster.update(np.random.normal(size=(2, 1)))
```

**Comment by fkiraly** [comment]
well, it would be great if you could look at the proposed fix in #1365.
